shell/utils/unit-tests/ChildRenderingRouterLinkStub.ts

```diff
@@ -0,0 +1,36 @@
+import { RouterLinkStub } from '@vue/test-utils';
+import { NavigationFailure, Route } from 'vue-router';
+
+/**
+ * See {@link RouterLinkSlotArgument} in vue-router
+ */
+export interface RouterLinkSlotArgumentOptional {
+    href?: string;
+    route?: Route;
+    navigate?: (e?: MouseEvent) => Promise<undefined | NavigationFailure>;
+    isActive?: boolean;
+    isExactActive?: boolean;
+}
+
+/**
+ * This is a workaround because VueUtils RouterLinkStub doesn't currently support the slot api.
```

Comment:
The aim of unit test is to separate the functionalities of the code into more parts, simplify the troubleshooting, and ensure to repeat the same tests.
By providing this utility it seems, please correct me if I am wrong, we are leading in the opposite direction.
Is there anything that needs to be inspected multiple times?

Reply:
Here's my definition of tests and unit tests just for understanding:

- Tests are meant to codify application invariants to ensure that changes in behavior do not go unnoticed.
- Unit tests are supposed to test small units (often functions or components) in isolation of the rest of the application

Because the vue test util's RouterLinkStub doesn't currently support the slots API we needed a way to test the Type component in isolation (without router-link). We also have other components that now use the slots API which we may want to add unit tests for. That makes me think this will allow us to improve and add more unit tests.

Ideally RouterLinkStub will add support for the slots API and we remove this.